pages/settings/interface.vue

```diff
@@ -1,3 +1,141 @@
+                        placeholder="/* Add your custom CSS here */
+:root {
+    --color-accent: theme('colors.pink.500');
+    --color-accent-hover: theme('colors.pink.400');
+}"
+                    ></textarea>
+                </div>
+                <div class="flex justify-end gap-x-3">
+                    <button
+                        @click="resetCustomCSS"
+                        class="rounded-md bg-zinc-800 px-3 py-2 text-sm font-semibold text-zinc-100 shadow-sm hover:bg-zinc-700"
+                    >
+                        Reset
+                    </button>
+                    <button
+                        @click="applyCustomCSS"
+                        class="rounded-md bg-blue-600 px-3 py-2 text-sm font-semibold text-white shadow-sm hover:bg-blue-500"
+                    >
+                        Apply
+                    </button>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<script setup lang="ts">
+import { SwatchIcon, ChevronRightIcon, CodeBracketIcon } from "@heroicons/vue/24/outline";
+
+const themes = [
+    { id: 'dark', name: 'Dark (Default)', bgColor: 'bg-zinc-950', palette: 'Zinc' },
+    { id: 'light', name: 'Light', bgColor: 'bg-white', palette: 'White/Gray' },
+    { id: 'theme-purple', name: 'Purple', bgColor: 'bg-purple-900', palette: 'Purple/Violet' },
+    { id: 'theme-green', name: 'Green', bgColor: 'bg-emerald-900', palette: 'Emerald/Teal' },
+    { id: 'theme-ocean', name: 'Ocean', bgColor: 'bg-slate-900', palette: 'Slate/Cyan' },
+    { id: 'theme-amber', name: 'Amber', bgColor: 'bg-amber-900', palette: 'Amber/Orange' },
+    { id: 'theme-rose', name: 'Rose', bgColor: 'bg-rose-900', palette: 'Rose/Pink' },
+    { id: 'theme-indigo', name: 'Indigo', bgColor: 'bg-indigo-900', palette: 'Indigo/Violet' },
+];
+
+const { currentTheme, setTheme } = useTheme();
+const showAdvanced = ref(false);
+const customCSS = ref(localStorage.getItem('customCSS') || '');
+
+// Create style element for custom CSS
+const customStyleElement = ref<HTMLStyleElement | null>(null);
+
+function applyCustomCSS() {
+    // Store in localStorage
+    localStorage.setItem('customCSS', customCSS.value);
     
-</template>
\ No newline at end of file
+    // Create or update style element
+    if (!customStyleElement.value) {
+        customStyleElement.value = document.createElement('style');
+        customStyleElement.value.id = 'custom-css';
+        document.head.appendChild(customStyleElement.value);
+    }
+    customStyleElement.value.textContent = customCSS.value;
+}
+
+function resetCustomCSS() {
+    customCSS.value = '';
+    localStorage.removeItem('customCSS');
+    if (customStyleElement.value) {
+        customStyleElement.value.textContent = '';
+    }
+}
+
+// Initialize custom CSS on mount
+onMounted(() => {
+    if (customCSS.value) {
+        applyCustomCSS();
+    }
+});
```

Comment:
Instead of doing all this complicated and anti-pattern `document` logic, can we use `v-bind` in a Vue style tag?